pkg/ipcache/metadata.go

```diff
@@ -214,6 +216,43 @@ func (m *metadata) getLocked(prefix netip.Prefix) prefixInfo {
 	return m.m[canonicalPrefix(prefix)]
 }
 
+// findCIDRParentPrefix returns the closest parent prefix has a parent prefix with a CIDR label
+// in the metadata cache
+func (m *metadata) findCIDRParentPrefix(prefix netip.Prefix) (parent netip.Prefix, ok bool) {
+	for bits := prefix.Bits() - 1; bits > 0; bits-- {
+		parent, _ = prefix.Addr().Unmap().Prefix(bits) // canonical
+		if info, ok := m.m[parent]; ok && info.hasLabelSource(labels.LabelSourceCIDR) {
+			return parent, true
+		}
+	}
+
+	return netip.Prefix{}, false
+}
+
+func isChildPrefix(parent, child netip.Prefix) bool {
+	if child == parent {
+		return true
+	}
+
+	return parent.Contains(child.Addr()) && child.Bits() >= parent.Bits()
```

Comment:
Why is the latter part of this expression needed (the bits check)? Isn't `Contains()` enough?

Reply:
Child could have the same base address but a shorter prefix. So you have to check the length too.